easybuild/framework/easyblock.py

```diff
@@ -863,10 +860,22 @@ def make_builddir(self):
         # otherwise we wipe the already partially populated installation directory,
         # see https://github.com/easybuilders/easybuild-framework/issues/2556
         if not (self.build_in_installdir and self.iter_idx > 0):
+            # make sure we no longer sit in the build directory before cleaning it.
+            change_dir(self.orig_workdir)
             self.make_dir(self.builddir, self.cfg['cleanupoldbuild'])
 
         trace_msg("build dir: %s" % self.builddir)
 
+    def reset_changes(self):
+        """
+        Reset environment
+        """
+        env.reset_changes()
+        if self.reset_environ is None:
+            self.reset_environ = copy.deepcopy(os.environ)
```

Comment:
@bartoldeman Please add a comment to explain why keeping tracking of the environment in `reset_environ` is needed, since that's not very obvious here...